test: add ansible 2.16

sheldonhull opened the pull request:
- Fix magefile to also handle release fragment missing.
- Generate latest release version.

sheldonhull committed:
tests: add ansible 2.16

- Fix magefile to also handle release fragment missing.
- Generate latest release version.

sheldonhull committed:
chore: update dependencies for tooling

sheldonhull committed:
chore(ci): update yaml files

- Replace double quotes with single quotes in `.aqua/aqua.yaml`
- Update Python version specification format in `.github/workflows/test.yaml`
- Remove unnecessary `working-directory` specification in `.github/workflows/test.yaml`

sheldonhull committed:
chore(ci): update test workflow in GitHub Actions

- Switch branch and directory name quotes in test.yaml to single quotes
- Update aqua_version in aqua action from v1.32.0 to v2.21.3
- Add 'update-github-path-with-aqua' and 'aqua-install-tooling' steps
- Change aqua_opts in both Aqua steps to '--tags installfirst' and '--tags tests' respectively

sheldonhull committed:
chore(ci): update python version and modify working directory in pipeline

- Switch Python version from 3.10 to 3.11 in pipeline
- Modify GitHub workflow to consistently set working-directory as './ansible_collections/delinea/core'
- Change double quoted strings in YAML file to single quoted

sheldonhull committed:
chore(ci): update test.yaml with minor changes

- switch to single quotes for strings in 'branches' and 'HERE' values
- downgrade python-version to '3.10'
- replace double quotes with single in 'aqua_opts' value

andrii-zakurenyi commented on .github/workflows/test.yaml:
```diff
@@ -26,6 +26,7 @@ jobs:
           - stable-2.13
```
please drop 2.13 since it has reached EOL on 06 Nov 2023

andrii-zakurenyi replied:
![image](https://github.com/DelineaXPM/ansible-core-collection/assets/85106843/62148636-7ab5-43b3-b6a6-af5b27b741a4)
link: https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#ansible-core-support-matrix

sheldonhull replied:
done

sheldonhull committed:
chore(ci): modify github actions workflow

- Switched to single quotes for string values
- Removed `stable-2.13` from ansible version matrix
- Unchanged python version and aqua options display formatting
